apps/examples/rpc_demo/rpc_demod.c

```diff
@@ -35,33 +36,27 @@
 static void *platform;
 static struct rpmsg_device *rpdev;
 static struct rpmsg_endpoint app_ept;
+static struct metal_io_region *shbuf_io;
 static int request_termination = 0;
 static int ept_deleted = 0;
 static int err_cnt = 0;
 
-static void *copy_from_shbuf(void *dst, void *shbuf, int len)
+static int copy_from_shbuf(void *dst, void *shbuf, int len)
 {
-	void *retdst = dst;
+	int ret;
+	unsigned long offset = metal_io_virt_to_offset(shbuf_io, shbuf);
 
-	while (len && ((uintptr_t)shbuf % sizeof(int))) {
-		*(uint8_t *)dst = *(uint8_t *)shbuf;
-		dst++;
-		shbuf++;
-		len--;
-	}
-	while (len >= (int)sizeof(int)) {
-		*(unsigned int *)dst = *(unsigned int *)shbuf;
-		dst += sizeof(int);
-		shbuf += sizeof(int);
-		len -= sizeof(int);
-	}
-	while (len > 0) {
-		*(uint8_t *)dst = *(uint8_t *)shbuf;
-		dst++;
-		shbuf++;
-		len--;
+	if (offset == METAL_BAD_OFFSET) {
+		LPERROR("no offset within IO region for data ptr: %x \r\n",
```

Comment:
build warning here:
```
/local/home/frq07267/views/openamp/open-amp/apps/examples/rpc_demo/rpc_demod.c:33:41: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 2 has type ‘void *’ [-Wformat=]
   33 | #define LPRINTF(format, ...) raw_printf("Host> " format, ##__VA_ARGS__)
      |                                         ^~~~~~~~
/local/home/frq07267/views/openamp/open-amp/apps/examples/rpc_demo/rpc_demod.c:32:40: note: in definition of macro ‘raw_printf’
   32 | #define raw_printf(format, ...) printf(format, ##__VA_ARGS__)
      |                                        ^~~~~~
/local/home/frq07267/views/openamp/open-amp/apps/examples/rpc_demo/rpc_demod.c:34:30: note: in expansion of macro ‘LPRINTF’
   34 | #define LPERROR(format, ...) LPRINTF("ERROR: " format, ##__VA_ARGS__)
      |                              ^~~~~~~
/local/home/frq07267/views/openamp/open-amp/apps/examples/rpc_demo/rpc_demod.c:50:3: note: in expansion of macro ‘LPERROR’
   50 |   LPERROR("no offset within IO region for data ptr: %x \r\n",
      |   ^~~~~~~
/local/home/frq07267/views/openamp/open-amp/apps/examples/rpc_demo/rpc_demod.c:50:54: note: format string is defined here
   50 |   LPERROR("no offset within IO region for data ptr: %x \r\n",
      |                                                     ~^
      |                                                      |
      |                                                      unsigned int
      |                                                     %p

```

Reply:
Sorry, I missed it. That's embarrassing. Thank you for catching it.